### PR TITLE
Tmemowjb

### PR DESCRIPTION
**Pull request by wjblanke**
thread safe tables for FSE

### CODE REVIEW

**Review by hoffmang9 — APPROVED** [review]
+1

File: src/encoding.hpp [hunk 1]
```diff
@@ -29,8 +29,68 @@
...
-std::map<double, FSE_CTable *> CT_MEMO = {};
-std::map<double, FSE_DTable *> DT_MEMO = {};
+#include <mutex>
+
+class TMemoCache {
+public:
+    ~TMemoCache() 
+    {
+        // Clean up global entries on destruction
+        std::lock_guard<std::mutex> l(memoMutex);
```
Comment (arvidn) [inline src/encoding.hpp:39]: this looks a bit suspicious. By the time this object is being destructed, no other thread has better hold a reference to it. Is this actually protecting against something that can happen?

Or was this mutex meant to be `static` as well, so multiple `TMemoCache` objects were meant to share it?

Comment (wjblanke) [reply src/encoding.hpp]: I don't think it is needed, I just had it to mirror the other functions. Its gone now.

**Review by hoffmang9 — APPROVED** [review]
heh on varname length

**Pull request merged**